File: backend/app/schemas/quality_metrics.py
from pydantic import BaseModel


class QualityMetricsResponse(BaseModel):
    total_ai_responses: int
    total_tickets_analyzed: int
    total_feedback: int
    average_rating: float | None
    helpful_count: int
    not_helpful_count: int
    feedback_coverage_percent: float
    rating_distribution: dict[str, int]
    responses_without_feedback: int

File: frontend/src/types/qualityMetrics.ts
export interface QualityMetrics {
  total_ai_responses: number
  total_tickets_analyzed: number
  total_feedback: number
  average_rating: number | null
  helpful_count: number
  not_helpful_count: number
  feedback_coverage_percent: number
  rating_distribution: Record<string, number>
  responses_without_feedback: number
}
